### ISSUE TITLE
Fire Delay Disobeys 

### ISSUE THREAD

**Issue by Fox-McCloud** [ParadiseSS13/Paradise#433]
Bit pre-emptive, but.

Fire delay behaves rather oddly with multimode guns:

-If you shoot a projectile that has a delay, then switch to a new mode, you can start firing that new mode's projectile immediately
-If you shoot a projectile that as a delay, switch to a new mode and leave it in that mode for a long time, then switch back, it'll still make you wait for the fire delay cooldown to occur; the cooldown seems to only tick down when it's actively in that mode

basically, it seems fire delay and modes are independent of each other, when they probably shouldn't be.

**Comment by Markolie** [comment]
Rather than having one, central proc for mode switching, every gun with several modes uses its own system, which changes the fire_delay variable so it takes longer to, say, fire a taser shot than a disabler. The disabler has a fire_delay of 0, so can constantly fire. The taser has a delay of 20, so it takes a while until you can shoot. While it may seem like every mode has an independent fire handler, it really doesn't. Grab an M1911-P, fire a taser, switch to a different mode, wait two seconds, switch back to taser and you can fire. If you switch between modes quickly enough it'll seem as though the cooldown for tasers doesn't count down until you switch to the mode, but that's simply because the fire_delay is increased.

Fire delay is handled by ready_to_fire in gun.dm, which isn't overriden anywhere. It checks last_fired, which isn't overridden anywhere either. So, this isn't really a bug.

**Comment by Markolie** [comment]
derp, messing up the tags
